### PR TITLE
Improve Build nodes2way DB Performance

### PR DESCRIPTION
**Pull request by wangyoucao577**
# Issue

Closes https://github.com/Telenav/osrm-backend/issues/272

### Improvement    
- Write DB by `BatchWrite` instead of `Write`;
- Read `csv` by `csvreader`(improved read csv);
- Parallel parsing csv lines to wayID,nodeIDs;

A significant improvement(about 50%) has been made by the changes, espically for huge files.     
It still needs a lot of time since the build `B+` tree is very heavy and can not in parallel.       

### Detailed Benchmark and Profiling 
- [272 issuecomment - nodes2way-builder Benchmark](https://github.com/Telenav/osrm-backend/issues/272#issuecomment-614022269)
- [272 issuecomment - Extract - Read data from csv](https://github.com/Telenav/osrm-backend/issues/272#issuecomment-613998609)
- [272 issuecomment - Load - Write data into DB - Profiling
](https://github.com/Telenav/osrm-backend/issues/272#issuecomment-612877931)

### CODE REVIEW

File: integration/util/csvreader/csv_records_reader.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+// It returns minimum MinReadCount csv records(except last read) and true per call if data available.
+// If the second value returns false, the async reading process has been closed, and you'd better to check whether any Err() occurs.
+// It's safe to call it by multiple goroutines at the same time.
+func (r *RecordsAsyncReader) ReadRecords() ([][]string, bool) {
+	records, ok := <-r.recordsChan
+	return records, ok
+}
+
+func (r *RecordsAsyncReader) parser() {
+
```
Comment (CodeBear801) [inline integration/util/csvreader/csv_records_reader.go:62]: parse()

**Comment by CodeBear801** [comment]
Great work and detailed profile result.  From your profile, it seems that do need to be aware of performance drop of too many small items put into channel.  

Here is my understanding of changes:

For input csv
- read compressed input(by snappy)
- patch csv lines by `MinReadCount` and put patched result into channel not single line
- 500 lines -> 500 records of way -> nodeids

For batch write
- each time get 500 records from input channel -> batch write 100 records

File: integration/util/csvreader/csv_records_reader.go (hunk 1 above)
Comment (wangyoucao577) [reply integration/util/csvreader/csv_records_reader.go:62]: done.

**Comment by wangyoucao577** [comment]
> Great work and detailed profile result. From your profile, it seems that do need to be aware of performance drop of too many small items put into channel.
> 
> Here is my understanding of changes:
> 
> For input csv
> 
> * read compressed input(by snappy)
> * patch csv lines by `MinReadCount` and put patched result into channel not single line
> * 500 lines -> 500 records of way -> nodeids
> 
> For batch write
> 
> * each time get 500 records from input channel -> batch write 100 records

Absolutely! Only a small difference: read raw csv input since it's a preprocess and we do have the csv before snappy compress.

**Pull request merged**